frontend/app/src/components/home/AccountInfo.svelte

```diff
@@ -26,19 +33,90 @@
         fullWidthOnMobile = false,
     }: Props = $props();
 
+    const client = getContext<OpenChat>("client");
+
     let tokenDetails = $derived($cryptoLookup.get(ledger)!);
-    let account = $derived(
-        tokenDetails.symbol === ICP_SYMBOL ? $currentUserStore.cryptoAccount : $currentUserIdStore,
-    );
-    let symbol = $derived(tokenDetails.symbol);
+    let selectedNetwork = $state<string | undefined>(undefined);
+    let networks = $derived.by(() => {
+        if (tokenDetails.symbol === BTC_SYMBOL) {
+            return [BTC_SYMBOL, CKBTC_SYMBOL];
+        } else {
+            return [];
+        }
+    });
+    let btcAddress = $state<string | undefined>();
+
+    // Whenever the networks list changes, autoselect the first one
+    $effect(() => {
+        selectedNetwork = networks[0];
+    });
+
+    let account = $derived.by(() => {
+        if (tokenDetails.symbol === ICP_SYMBOL) {
+            return $currentUserStore.cryptoAccount;
+        } else if (tokenDetails.symbol === BTC_SYMBOL && selectedNetwork === BTC_SYMBOL) {
+            return btcAddress;
+        } else {
+            return $currentUserIdStore;
+        }
+    });
+
+    // We use `Lazy` to ensure this is only called once
+    const btcAddressPromise = new Lazy(() => client.getBtcAddress());
+
+    let error = $state();
+    $effect(() => {
+        if (account === undefined) {
+            if (selectedNetwork === BTC_SYMBOL) {
+                btcAddressPromise.get().then((addr) => btcAddress = addr).catch((e) => error = e);
```

Comment:
Isn't it fine to just call `client.getBtcAddress` directly here and not use `Lazy` at all? because internally `client.getBtcAddress` already deals with not loading the address more than once.

Reply:
It's quite a slow call as it goes User canister -> ckBTC canister (on another subnet).
So if you switch tokens back and forth it would call it twice.

Reply:
All the more reason to deal with it inside the OC client?